### PR TITLE
zsh-zhooks: init at 0-unstable-10-31-2021

### PR DESCRIPTION
**Pull request by fidgetingbits**
## Description of changes

Add myself to maintainers list and add a new simple zsh plugin as my first package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/shells/zsh/zhooks/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ lib, stdenv, fetchgit }:
+let
+  pname = "zhooks";
+  install_path = "share/zsh/${pname}";
+in
+stdenv.mkDerivation {
+  inherit pname;
+  version = "e6616b4a2786b45a56a2f591b79439836e678d22";
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zhooks/default.nix:9]: This must use fetchFromGitHub

File: pkgs/shells/zsh/zhooks/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ lib, stdenv, fetchgit }:
+let
+  pname = "zhooks";
+  install_path = "share/zsh/${pname}";
+in
+stdenv.mkDerivation {
+  inherit pname;
+  version = "e6616b4a2786b45a56a2f591b79439836e678d22";
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zhooks/default.nix:8]: Please see the readme file in pkgs how to properly name version packages fetched from commits

File: pkgs/shells/zsh/zhooks/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchgit {
+    url = "https://github.com/agkozak/zhooks";
+    hash = "sha256-zahXMPeJ8kb/UZd85RBcMbomB7HjfEKzQKjF2NnumhQ=";
+  };
+  strictDeps = true;
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zhooks/default.nix:13]: There is no point in strictDeps as there aren't any other deps here

File: pkgs/shells/zsh/zhooks/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      To install the ${pname} plugin in home-manager you can add the following to your `programs.zsh.plugins` list:
+
+      ```nix
+      programs.zsh.plugins = [
+        {
+          name = "${pname}";
+          src = "''${pkgs.${pname}}/${install_path}";
+        }
+      ];
+      ```
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zhooks/default.nix:34]:
```suggestion
```

File: pkgs/shells/zsh/zhooks/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  };
+  strictDeps = true;
+  dontBuild = true;
+  installPhase = ''
+    install -m755 -D ${pname}.plugin.zsh --target-directory $out/${install_path}
+  '';
+  meta = with lib; {
+    homepage = "https://github.com/agkozak/zhooks";
+    license = licenses.mit;
+    longDescription = ''
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zhooks/default.nix:21]: A short description should be set, too

**Comment by fidgetingbits** [comment]
Thanks! All of the suggestion have been fixed. 

One thing I remembered is that things should go into `pkgs/by-name/` now so I also moved it and renamed it zsh-zhooks to match the naming of most other zsh plugins. Let me know if there are any concerns with that.

I also just remembered there is a push to no use `with` anymore, so also tweaked that.

**Comment by fidgetingbits** [comment]
@SuperSandro2000 can you take another review pass if you find the time?

**Comment by DontEatOreo** [comment]
Your commit is breaking [Commit Conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions). Please rebase.

In your case reword 215bad7cd9ad6cadff14e2d01fe6aab9827ae6e2 to `zhook: init at 0-unstable-10-31-2021` and squash 883e162c441601905d45bc37a6fd27f32c9c5db6 5cb02ebb8ac4ee6eb5fe37e4998c1b93cd2cb918 into 215bad7cd9ad6cadff14e2d01fe6aab9827ae6e2

File: pkgs/by-name/zs/zsh-zhooks/package.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    owner = "agkozak";
+    repo = "zhooks";
+    rev = "e6616b4a2786b45a56a2f591b79439836e678d22";
+    sha256 = "sha256-zahXMPeJ8kb/UZd85RBcMbomB7HjfEKzQKjF2NnumhQ=";
+  };
+  dontBuild = true;
+  installPhase = ''
+    install -m755 -D ${pname}.plugin.zsh --target-directory $out/${install_path}
+  '';
+  meta = {
```
Comment (DontEatOreo) [inline pkgs/by-name/zs/zsh-zhooks/package.nix:19]:
```suggestion
  pname = "zsh-${pname}";
  version = "0-unstable-10-31-2021";

  src = fetchFromGitHub {
    owner = "agkozak";
    repo = "zhooks";
    rev = "e6616b4a2786b45a56a2f591b79439836e678d22";
    sha256 = "sha256-zahXMPeJ8kb/UZd85RBcMbomB7HjfEKzQKjF2NnumhQ=";
  };

  dontBuild = true;

  installPhase = ''
    install -m755 -D ${pname}.plugin.zsh --target-directory $out/${install_path}
  '';

  meta = {
```

Just a small nit

File: pkgs/by-name/zs/zsh-zhooks/package.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, fetchFromGitHub }:
```
Comment (DontEatOreo) [inline pkgs/by-name/zs/zsh-zhooks/package.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
}:
```

Formatted as per RFC 0166 with package `nixfmt-rfc-style`

File: pkgs/by-name/zs/zsh-zhooks/package.nix [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  meta = {
+    homepage = "https://github.com/agkozak/zhooks";
+    license = lib.licenses.mit;
+    description = "A tool for displaying the code for all Zsh hook functions";
+    longDescription = ''
+      This Zsh plugin is a tool for displaying the code for all Zsh hook functions (such as precmd), as well as the contents of
+      hook arrays (such as precmd_functions).
+      ```
+    '';
+    maintainers = [ lib.maintainers.fidgetingbits ];
```
Comment (DontEatOreo) [inline pkgs/by-name/zs/zsh-zhooks/package.nix:28]:
```suggestion
    description = "A tool for displaying the code for all Zsh hook functions";
    homepage = "https://github.com/agkozak/zhooks";
    license = lib.licenses.mit;
    longDescription = ''
      This Zsh plugin is a tool for displaying the code for all Zsh hook functions (such as precmd), as well as the contents of
      hook arrays (such as precmd_functions).
      '';
    maintainers = [ lib.maintainers.fidgetingbits ];
```

Alphabetical order

**Review by 4evy — CHANGES_REQUESTED** [review]

**Comment by DontEatOreo** [comment]
Please rename your PR title to `zhook: init at 0-unstable-10-31-2021` as well

File: pkgs/by-name/zs/zsh-zhooks/package.nix (hunk 7 above)
Comment (fidgetingbits) [reply pkgs/by-name/zs/zsh-zhooks/package.nix]: I ran `nix run nixpkgs-unstable#nixfmt-rfc-style -- package.nix` with a fully updated unstable and the comma placement seems different than what you are suggesting. I've left it as it was from me running the tool, but if this is somehow wrong let me know.

**Comment by fidgetingbits** [comment]
Thanks @DontEatOreo. Just noted the one difference between our nixfmt-rfc-style formatting, but otherwise should all be addressed.

**Review by 4evy — APPROVED** [review]

File: pkgs/by-name/zs/zsh-zhooks/package.nix (hunk 7 above)
Comment (DontEatOreo) [reply pkgs/by-name/zs/zsh-zhooks/package.nix]: Good catch, sorry about that!

File: pkgs/by-name/zs/zsh-zhooks/package.nix [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+      owner = "agkozak";
+      repo = "zhooks";
+      rev = "e6616b4a2786b45a56a2f591b79439836e678d22";
+      sha256 = "sha256-zahXMPeJ8kb/UZd85RBcMbomB7HjfEKzQKjF2NnumhQ=";
+    };
+
+    dontBuild = true;
+
+    installPhase = ''
+      install -m755 -D ${pname}.plugin.zsh --target-directory $out/${install_path}
```
Comment (SuperSandro2000) [inline pkgs/by-name/zs/zsh-zhooks/package.nix:23]:
```suggestion
      install -m755 -D zhooks.plugin.zsh --target-directory $out/share/zsh/zhooks
```

File: pkgs/by-name/zs/zsh-zhooks/package.nix [hunk 10]
```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}: let
+  pname = "zhooks";
+  install_path = "share/zsh/${pname}";
+in
+  stdenv.mkDerivation {
+    pname = "zsh-${pname}";
```
Comment (SuperSandro2000) [inline pkgs/by-name/zs/zsh-zhooks/package.nix:10]:
```suggestion
    pname = "zsh-zhooks";
```
declaring pname twice can easily lead to errors. Lets just write them out

File: pkgs/by-name/zs/zsh-zhooks/package.nix [hunk 11]
```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}: let
+  pname = "zhooks";
+  install_path = "share/zsh/${pname}";
+in
+  stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/by-name/zs/zsh-zhooks/package.nix:9]:
```suggestion
}:

stdenv.mkDerivation {
```

File: pkgs/by-name/zs/zsh-zhooks/package.nix (hunk 10 above)
Comment (fidgetingbits) [reply pkgs/by-name/zs/zsh-zhooks/package.nix]: good catch thanks, fixed

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 296562` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zsh-zhooks</li>
  </ul>
</details>

**Review by JohnRTitor — APPROVED** [review]
LGTM

**Pull request merged**